File: python/9_isPalindrome.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        if x < 0: return False
        if x < 10: return True
        
        y, tmp = 0, x

        while tmp:
            y = 10 * y + tmp % 10
            tmp //= 10
        return x == y

File: C/9_isPalindrome.c
bool isPalindrome(int x) {
    int temp = x;
    int reverse = 0;
    
    if (x < 0)
        return false;

    if (x < 10)
        return true;
    
    while (temp) {
        reverse = reverse * 10 + temp % 10;
        temp /= 10;
    }
    
    return x == reverse;
}
